### PR TITLE
Reworked mapping between Board and BoardColumn

### PR DESCRIPTION
**Pull request by ChernykhVyacheslav**
https://trello.com/c/M8KmQcH6/195-rework-entity-mapping-for-board-and-boardcolumn

### CODE REVIEW

**Review by jules-n — APPROVED** [review]
Looks fine

File: service/src/main/java/com/softserve/borda/services/impl/BoardColumnServiceImpl.java [hunk 1]
```diff
@@ -52,31 +49,14 @@ public boolean deleteBoardColumnById(Long id) {
...
-            board.getBoardColumns().remove(boardColumn);
-            deleteBoardColumnById(boardColumn.getId());
-            boardService.update(board);
-            return true;
-        } catch (Exception e) {
-            log.severe(e.getMessage());
-            return false;
-        }
+        boardColumn.setBoardId(boardId);
+        return update(boardColumn);
```
Comment (achehrynets) [inline service/src/main/java/com/softserve/borda/services/impl/BoardColumnServiceImpl.java]: It can be much easier, any ideas how we can refactor update in this case?
Hint: you can avoid update of all attributes of a board column and even avoid select of  board column

File: controller/src/main/java/com/softserve/borda/controllers/BoardController.java [hunk 2]
```diff
@@ -454,21 +454,19 @@ public Board getBoard(@PathVariable Long id) {
     }
 
     @PostMapping("/users/invitations/{receiverUsername}/boards/{boardId}/roles/{userBoardRoleId}")
-
     public ResponseEntity<Object> createInvitation(Authentication authentication,
                                                           @PathVariable String receiverUsername,
                                                           @PathVariable Long boardId,
                                                           @PathVariable Long userBoardRoleId) {
-        var user = userService.getUserByUsername(authentication.getName());
+        User sender = userService.getUserByUsername(authentication.getName());
```
Comment (achehrynets) [inline controller/src/main/java/com/softserve/borda/controllers/BoardController.java:461]: Do not make changes that are not related to a task.
it can be a cause of merge conflicts.

**Review by achehrynets — COMMENTED** [review]
Left comments

File: service/src/main/java/com/softserve/borda/services/impl/BoardColumnServiceImpl.java (hunk 1 above)
Comment (ChernykhVyacheslav) [reply service/src/main/java/com/softserve/borda/services/impl/BoardColumnServiceImpl.java]: We can just user repository method save() . To avoid select, I guess we can create custom method in repository that will only update one field. Will try to do it now

File: controller/src/main/java/com/softserve/borda/controllers/BoardController.java (hunk 2 above)
Comment (ChernykhVyacheslav) [reply controller/src/main/java/com/softserve/borda/controllers/BoardController.java:461]: Yes, my bad. There was some error after merge and after fixing it I also refactored it a bit. Will try not to do it in the future.

File: persistence/src/main/java/com/softserve/borda/repositories/BoardColumnRepository.java [hunk 3]
```diff
@@ -2,8 +2,20 @@
...
+import java.util.List;
 
 @Repository
 public interface BoardColumnRepository extends JpaRepository<BoardColumn, Long> {
+    List<BoardColumn> getAllBoardColumnsByBoardId(Long boardId);
+
+    @Transactional
+    @Modifying
+    @Query("update board_columns board_column set board_column.boardId = :boardId where board_column.id = :boardColumnId")
+    Integer setBoardIdForBoardColumn(@Param("boardColumnId") Long boardColumnId, @Param("boardId") Long boardId);
```
Comment (achehrynets) [inline persistence/src/main/java/com/softserve/borda/repositories/BoardColumnRepository.java]: you can do only CRUD operations on repository level
C - create
R - read/find/select
U - update
D - delete/remove

I think it is good to use `update` instead of `set` keyword here.
Same case applies for method above.

File: persistence/src/main/java/com/softserve/borda/repositories/BoardColumnRepository.java [hunk 4]
```diff
@@ -2,8 +2,20 @@
...
 import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
 
 @Repository
 public interface BoardColumnRepository extends JpaRepository<BoardColumn, Long> {
+    List<BoardColumn> getAllBoardColumnsByBoardId(Long boardId);
+
+    @Transactional
```
Comment (achehrynets) [inline persistence/src/main/java/com/softserve/borda/repositories/BoardColumnRepository.java]: I think, it should be on service layer
More info you can find there - https://habr.com/ru/post/532000/

**Review by achehrynets — COMMENTED** [review]
Left comments, once they are resolved you can merge PR.

File: persistence/src/main/java/com/softserve/borda/repositories/BoardColumnRepository.java (hunk 3 above)
Comment (ChernykhVyacheslav) [reply persistence/src/main/java/com/softserve/borda/repositories/BoardColumnRepository.java]: Got it, thanks. Fixed both

File: persistence/src/main/java/com/softserve/borda/repositories/BoardColumnRepository.java (hunk 4 above)
Comment (ChernykhVyacheslav) [reply persistence/src/main/java/com/softserve/borda/repositories/BoardColumnRepository.java]: Thanks, fixed. Will read info
